app/javascript/components/pages/Home.js

```diff
@@ -1,10 +1,173 @@
 import React, { Component } from 'react'
+import { UncontrolledCarousel, Row, Col, Card, CardBody, CardTitle, CardText, CardGroup, CardSubtitle, CardImg, Button } from "reactstrap";
 
 class Home extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      carouselPics : [
+        {
+          src: 'https://images.unsplash.com/photo-1501785888041-af3ef285b470?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80',
+          altText: 'Slide 1',
+          caption: 'Paradise',
+          key: 1,
+        },
+        {
+          src: 'https://images.unsplash.com/photo-1506197603052-3cc9c3a201bd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80',
+          altText: 'Slide 2',
+          caption: 'A Dream',
+          key: 2,
+        },
+        {
+          src: 'https://images.unsplash.com/photo-1446160657592-4782fb76fb99?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2069&q=80',
+          altText: 'Slide 3',
+          caption: 'Explore',
+          key: 3,
+        },
+        {
+          src: 'https://images.unsplash.com/photo-1436491865332-7a61a109cc05?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2074&q=80',
+          altText: 'Slide 4',
+          caption: 'Fantasy',
+          key: 4,
+        },
+        {
+          src: 'https://images.unsplash.com/photo-1520645521318-f03a712f0e67?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80',
+          altText: 'Slide 5',
+          caption: 'Travel',
+          key: 5,
+        },
+      ],
+      is_visible: false
+    }
+  }
+
+  componentDidMount(){
+    var scrollComponent = this;
+    document.addEventListener("scroll", function(e) {
+      scrollComponent.toggleVisibility()
+    })
+  }
+
+  toggleVisibility(){
+    if(window.pageYOffset > 300){
+      this.setState({is_visible: true})
+    } else {
+      this.setState({is_visible: false})
+    }
+  }
+
+  scrollToTop(){
+    window.scrollTo({top:0, behavior:"smooth"})
+  }
```

Comment:
I would be careful when utilizing document and window directly inside React as this is directly manipulating the DOM which is not best practice.